### ISSUE TITLE
log object is mutated on #stream

### ISSUE THREAD

**Issue by kleinron** [4umfreak/fun-bunyan#7]
fun-bunyan can be used to create an explicit stream.
when we create an explicit stream, like so:
```js
const bunyan = require('bunyan');
const funBunyan = require('fun-bunyan');

const streams = [
    // Log to Stackdriver or other...
    // stackdriver.stream(options.level || 'info');
];

if (process.env.NODE_ENV === 'development') {
    streams.push({
        stream: funBunyan.stream(),
        type: "raw",
        level: "info",
    });
}
```
the stream created by `funBunyan.stream()` - actually mutates the original log object (the entire log entry).
this has an unexpected side effect on other streams that need some of the fields being deleted by the stream.

**Comment by mvoorberg** [comment]
Interesting, I hadn't considered that. Can you provide a test / gist or example?

The following test appears to indicate that the original object is unaffected by the deletes...
```javascript
        it('doesn\'t mutate the log object', function (done) {
            // Deletes  ["name", "hostname", "pid", "msg", "err", "level", "time", "v"]
            const log = {
                name: "My App",
                hostname: "Espresso",
                message: "This is a log message."
            };
            funBunyan.error(log);
            expect(log.name).to.equal("My App");
            expect(log.hostname).to.equal("Espresso");
            expect(log.message).to.equal("This is a log message.");
            done();
        });
```
Cheers!

**Comment by kleinron** [comment]
the test you provided passes, but note that you use the "higher-level" class `funBunyan`, while I was referring to the "lower-level" writable `stream`.
the following test fails:
```js
it('doesn\'t mutate the log object', function (done) {
    const myStream = stream();
    // Deletes  ["name", "hostname", "pid", "msg", "err", "level", "time", "v"]
    const log = {
        name: "My App",
        v: 1,
        time: new Date(),
        level: 30,
        hostname: "Espresso",
        msg: "This is a log message."
    };
    myStream.write(log);
    expect(log.name).to.equal("My App");
    expect(log.hostname).to.equal("Espresso");
    expect(log.msg).to.equal("This is a log message.");
    done();
});
```

**Comment by mvoorberg** [comment]
Got it, thank-you!

Published v1.1.0 @ https://www.npmjs.com/package/fun-bunyan
